apph-back/src/test/java/com/viseo/apph/UserTest.java

```diff
@@ -37,85 +32,42 @@ public class UserTest {
     TypedQuery<Long> existByLoginQuery;
     @Mock
     PasswordEncoder passwordEncoder;
+   @Mock
+    com.viseo.apph.security.Utils utils;
 
     UserService userService;
     UserController userController;
+    JwtUtils jwtUtils;
 
     private void createUserController() {
         UserDao userDao = new UserDao();
         inject(userDao, "em", em);
         userService = new UserService();
         inject(userService, "userDao", userDao);
+        jwtUtils = new JwtUtils();
+        inject(userService, "jwtUtils", jwtUtils);
         userController = new UserController();
         inject(userController, "userService", userService);
         inject(userService, "encoder", passwordEncoder);
-    }
+        inject(userController, "utils", utils);
 
-    void inject(Object component, String field, Object injected) {
-        try {
-            Field compField = component.getClass().getDeclaredField(field);
-            compField.setAccessible(true);
-            compField.set(component, injected);
-        } catch (IllegalAccessException | NoSuchFieldException e) {
-            e.printStackTrace();
-        }
     }
 
     @Test
     public void testGetUserInfo() {
         //GIVEN
         createUserController();
         User user = new User().setLogin("toto").setPassword("password");
-        String jws = Jwts.builder().claim("login", user.getLogin()).setExpiration(new Date(System.currentTimeMillis() + 20000)).signWith(JwtConfig.getKey()).compact();
+        when(utils.getUser()).thenReturn(user);
         when(em.createQuery("SELECT u FROM User u WHERE u.login=:login", User.class)).thenReturn(userTypedQuery);
         when(userTypedQuery.getSingleResult()).thenReturn(new User().setLogin("toto").setPassword("password").setFirstname("firstname").setLastname("lastname"));
         when(userTypedQuery.setParameter("login", "toto")).thenReturn(userTypedQuery);
         //WHEN
-        ResponseEntity responseEntity = userController.getUserInfo(jws);
+        ResponseEntity responseEntity = userController.getUserInfo();
         //THEN
         Assert.assertTrue(responseEntity.getStatusCode().is2xxSuccessful());
     }
 
-    @Test
-    public void testFailUserNotFind() {
-        //GIVEN
-        createUserController();
-        String jws = Jwts.builder().claim("login", "dumb_toto").setExpiration(new Date(System.currentTimeMillis() + 20000)).signWith(JwtConfig.getKey()).compact();
-        //WHEN
-        ResponseEntity responseEntity = userController.getUserInfo(jws);
-        //THEN
-        Assert.assertEquals(HttpStatus.FORBIDDEN, responseEntity.getStatusCode());
-    }
```

Comment:
et ici tu décide de complètement supprimer le test, alors que tu as garder les deux derniers. . .

Reply:
C'est parce l'user ne peut pas être not find car sinon cela signifierait que l'user n'est pas authentifié et que donc la requete serait rejeté par le filtre de spring security. Donc ce test n'a pas lieu d'être

Reply:
Alors pourquoi avoir gardé les deux autres tests de user not found ?

Reply:
J'ai pas fait attention, je vais les retirer